tests/test_spacerequests_cleaner.py

```diff
@@ -0,0 +1,304 @@
+"""Test spacerequests_cleaner.py"""
+
+import datetime
+from unittest.mock import MagicMock, call, create_autospec
+
+import pytest
+from kubernetes import client, config  # type: ignore
+from pytest import MonkeyPatch
+
+from clean_spacerequests import spacerequests_cleaner
+from clean_spacerequests.spacerequests_cleaner import (
+    TIME_FORMAT,
+    delete_namespace_requests,
+    get_old_namespace_requests,
+    get_tenant_namespaces,
+    load_config,
+)
+
+# pylint: disable=too-few-public-methods
+
+
+class TestLoadConfig:
+    """Test load_config"""
+
+    def test_load_config_from_cluster(self) -> None:
+        """Test load config as if running inside a pod"""
+        conf: MagicMock = create_autospec(config)
+        load_config(conf)
+        conf.load_incluster_config.assert_called_once_with()
+        conf.load_kube_config.assert_not_called()
+
+    def test_load_config_from_kube_config(self) -> None:
+        """Test load config as if running outside of the cluster"""
+        conf: MagicMock = create_autospec(config)
+        conf.ConfigException = config.ConfigException
+        conf.load_incluster_config.side_effect = config.ConfigException
+        load_config(conf)
+        conf.load_incluster_config.assert_called_once_with()
+        conf.load_kube_config.assert_called_once_with()
+
+
+def get_mock_with_name_attribute(name) -> MagicMock:
+    """Since name  is an argument to the Mock constructor, if we want mock object with
+    name attribute, we need to set it after the mock has been created"""
+    mock = MagicMock()
+    mock.name = name
+    return mock
+
+
+class TestGetTenantNamespaces:
+    """Test get_tenant_namespaces"""
+
+    @pytest.mark.parametrize(
+        ("items", "tenant_namespaces"),
+        [
+            pytest.param(MagicMock(items=[]), [], id="no namespaces"),
+            pytest.param(
+                MagicMock(
+                    items=[MagicMock(metadata=get_mock_with_name_attribute("foo"))]
+                ),
+                ["foo"],
+                id="one namespace",
+            ),
+            pytest.param(
+                MagicMock(
+                    items=[
+                        MagicMock(metadata=get_mock_with_name_attribute("foo")),
+                        MagicMock(metadata=get_mock_with_name_attribute("bar")),
+                    ]
+                ),
+                ["foo", "bar"],
+                id="multiple namespaces",
+            ),
+        ],
+    )
+    def test_get_tenant_namespaces(self, items, tenant_namespaces) -> None:
```

Comment:
I think we should add the argument's types to all the methods in the test

Reply:
done